### ISSUE TITLE
Fatal error after updating to 3.88.0

### ISSUE THREAD

**Issue by sandervb2** [pulumi/pulumi#14146]
### What happened?

After updating to 3.88.0 Pulumi CLI encounters fatal error

### Example

```
================================================================================
The Pulumi CLI encountered a fatal error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v3.88.0
Go Version:       go1.21.1
Go Compiler:      gc
Architecture:     amd64
Operating System: linux
Panic:            fatal: A failure has occurred: unexpected wire type <nil>

goroutine 1 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:24 +0x5e
main.panicHandler(0xc00279ff07)
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/main.go:35 +0x45
panic({0x251a320?, 0xc002880090?})
        /opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
gopkg.in/yaml%2ev3.handleErr(0xc00279e8b8)
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:294 +0x6d
panic({0x251a320?, 0xc002880090?})
        /opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
gopkg.in/yaml%2ev3.handleErr(0xc00279df60)
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:294 +0x6d
panic({0x251a320?, 0xc002880090?})
        /opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Failf({0x2ad658a?, 0x5?}, {0xc00279d570?, 0x41910b?, 0x2a?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/fail.go:32 +0xcb
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.unmarshalObject({0x0?, 0x0?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:492 +0x54a
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.unmarshalObject({0x2626a00?, 0xc0027b2450?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:468 +0x5b1
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.unmarshalObject({0x24e0e20?, 0xc00280c138?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:484 +0x6a5
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*object).UnmarshalYAML(0xc00279da30, 0xc0027b8560)
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:435 +0x5b
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*Value).UnmarshalYAML(0xc00280c018, 0xc00280c018?)
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/value.go:152 +0x28
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler(0xc0009f5110, 0xc00256e460, {0x7f1b415620a8, 0xc00280c018})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:379 +0x94
gopkg.in/yaml%2ev3.(*decoder).prepare(0xc00279dbc8?, 0x410593?, {0x2998d40?, 0xc00280c018?, 0x18?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:427 +0x21e
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0009f5110, 0xc00256e460, {0x2998d40?, 0xc00280c018?, 0xc0027de910?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:502 +0x2b2
gopkg.in/yaml%2ev3.(*decoder).mapping(0xc0009f5110, 0xc00256e1e0, {0x262c280?, 0xc0027b63f0?, 0xc0009272e8?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:848 +0x98e
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0009f5110, 0xc00256e1e0, {0x24e3dc0?, 0xc0027b63f0?, 0x7f1b88648d28?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:510 +0x3ea
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler.func1({0x24e3dc0?, 0xc0027b63f0?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:381 +0xe5
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*Map).UnmarshalYAML(0xc0022a91d0, 0xc0027b8520)
        /home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/map.go:239 +0x82
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler(0xc0009f5110, 0xc00256e1e0, {0x7f1b41562068, 0xc0022a91d0})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:379 +0x94
gopkg.in/yaml%2ev3.(*decoder).prepare(0x251a320?, 0xc00256e140?, {0x29251e0?, 0xc0022a91d0?, 0x251a320?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:427 +0x21e
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0009f5110, 0xc00256e1e0, {0x29251e0?, 0xc0022a91d0?, 0x9f980f?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:502 +0x2b2
gopkg.in/yaml%2ev3.(*decoder).mappingStruct(0xc0009f5110, 0xc00256e0a0, {0x2952ae0?, 0xc0022a91a0?, 0x40dc5f?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:935 +0xf74
gopkg.in/yaml%2ev3.(*decoder).mapping(0xc0009f5110, 0xc00256e0a0, {0x2952ae0?, 0xc0022a91a0?, 0x0?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:786 +0xa5
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0009f5110, 0xc00256e0a0, {0x2952ae0?, 0xc0022a91a0?, 0x9fa8a5?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:510 +0x3ea
gopkg.in/yaml%2ev3.(*decoder).document(...)
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:527
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0009f5110, 0xc00256e000, {0x2952ae0?, 0xc0022a91a0?, 0x0?})
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:498 +0x28a
gopkg.in/yaml%2ev3.unmarshal({0xc002470000, 0xd5e5, 0x10000}, {0x26f92a0, 0xc0022a91a0}, 0x58?)
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:167 +0x3b1
gopkg.in/yaml%2ev3.Unmarshal(...)
        /home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:89
github.com/pulumi/pulumi/sdk/v3/go/common/encoding.(*yamlMarshaler).Unmarshal(0xc0021f4370?, {0xc002470000?, 0xc002316e70?, 0x47eff60?}, {0x26f92a0?, 0xc0022a91a0?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/encoding/marshal.go:113 +0x39
github.com/pulumi/pulumi/sdk/v3/go/common/workspace.LoadProjectStack(0xc0020edb78?, {0xc002087c00, 0x3d})
        /home/runner/work/pulumi/pulumi/sdk/go/common/workspace/loaders.go:192 +0x1f9
github.com/pulumi/pulumi/sdk/v3/go/common/workspace.DetectProjectStack({0xc0020edb78?, 0x2a66e83?})
        /home/runner/work/pulumi/pulumi/sdk/go/common/workspace/paths.go:162 +0x30
main.loadProjectStack(0x2a5d883?, {0x329b3e0?, 0xc0020bf170?})
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/config.go:738 +0x3c
main.getStackConfigurationWithFallback({0x32830d0, 0x4874ea0}, {0x329b3e0, 0xc0020bf170}, 0x32830d0?, {0x0, 0x0}, 0x0)
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/config.go:1113 +0xa5
main.getStackConfiguration(...)
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/config.go:1024
main.newUpCmd.func1({_, _}, {{{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, 0x0, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/up.go:109 +0x309
main.newUpCmd.func3(0x0?, {0x4874ea0, 0x0, 0x0?})
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/up.go:513 +0xb8e
main.newUpCmd.RunResultFunc.func4(0xc000bb0100?, {0x4874ea0, 0x0, 0x0})
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/cmdutil/exit.go:115 +0x38
github.com/spf13/cobra.(*Command).execute(0xc000bae600, {0x4874ea0, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0xc001a51200)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/main.go:56 +0x58
``` 

### Output of `pulumi about`

```
Backend        
Name           <backend-name>
URL            <s3-backend>
User           <user>
Organizations  
Token type     personal

Dependencies:
NAME        VERSION
boto3       1.28.53
fabric      3.2.2
pip         22.3.1
psutil      5.9.5
pulumi-aws  6.4.0
requests    2.31.0
setuptools  65.5.1
wheel       0.38.4

Pulumi locates its logs in /tmp by default

``` 

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tgummerer** [comment]
I now realize I could have done a better job updating this.  This issue seems to appear when there's an empty section in a yaml, that we previously accepted, but we now end up `panic`ing.

For example something like this:
```
config:
  aws:region: us-east-1
  aws:id:
    -
```

We're working on a fix for this in https://github.com/pulumi/pulumi/pull/14149, but as a workaround in the meantime please look for any such section in your yaml files, and remove those.  This should get pulumi back to working.

**Comment by mfzl** [comment]
We started having a similar issue after our github action started using the 3.88.0 We've downgraded Pulumi and is working for us now.

Here is the stack trace.

```
================================================================================
The Pulumi CLI encountered a fatal error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v3.88.0
Go Version:       go1.21.1
Go Compiler:      gc
Architecture:     amd64
Operating System: linux
Panic:            fatal: A failure has occurred: unexpected wire type time.Time

goroutine 1 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:24 +0x5e
main.panicHandler(0xc001c3ff07)
	/home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/main.go:35 +0x45
panic({0x251a320?, 0xc00202ceb0?})
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
gopkg.in/yaml%2ev3.handleErr(0xc001c3e8b8)
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:294 +0x6d
panic({0x251a320?, 0xc00202ceb0?})
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
gopkg.in/yaml%2ev3.handleErr(0xc001c3df60)
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/yaml.go:294 +0x6d
panic({0x251a320?, 0xc00202ceb0?})
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:914 +0x21f
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
	/home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Failf({0x2ad658a?, 0x69?}, {0xc001c3d730?, 0x6?, 0xc001c3d748?})
	/home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/fail.go:32 +0xcb
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.unmarshalObject({0x2a37340?, 0xc001ea78c0?})
	/home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:492 +0x54a
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.unmarshalObject({0x2626a00?, 0xc001bb1bc0?})
	/home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:468 +0x5b1
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*object).UnmarshalYAML(0xc001c3da30, 0xc001e9f900)
	/home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/object.go:435 +0x5b
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*Value).UnmarshalYAML(0xc001ea7890, 0xc001ea7890?)
	/home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/value.go:152 +0x28
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler(0xc0007b1500, 0xc001fe0b40, {0x7f2dd0b6ff98, 0xc001ea7890})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:379 +0x94
gopkg.in/yaml%2ev3.(*decoder).prepare(0xc001c3dbc8?, 0x410565?, {0x2998d40?, 0xc001ea7890?, 0x18?})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:427 +0x21e
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0007b1500, 0xc001fe0b40, {0x2998d40?, 0xc001ea7890?, 0xc002026b10?})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:502 +0x2b2
gopkg.in/yaml%2ev3.(*decoder).mapping(0xc0007b1500, 0xc001f75680, {0x262c280?, 0xc0000814b8?, 0x249a509?})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:848 +0x98e
gopkg.in/yaml%2ev3.(*decoder).unmarshal(0xc0007b1500, 0xc001f75680, {0x24e3dc0?, 0xc0000814b8?, 0x7f2e17aec5b8?})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:510 +0x3ea
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler.func1({0x24e3dc0?, 0xc0000814b8?})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:381 +0xe5
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.(*Map).UnmarshalYAML(0xc001e97710, 0xc001e9f420)
	/home/runner/work/pulumi/pulumi/sdk/go/common/resource/config/map.go:239 +0x82
gopkg.in/yaml%2ev3.(*decoder).callObsoleteUnmarshaler(0xc0007b1500, 0xc001f75680, {0x7f2dd0b6ff58, 0xc001e97710})
	/home/runner/go/pkg/mod/gopkg.in/yaml.v3@v3.0.1/decode.go:379 +0x94
gopkg.in/yaml%2ev3.(*decoder).prepare(0x251a320?, 0xc00
```

**Comment by andrewSC** [comment]
While I've fixed my `*.yaml` files using `yamllint` and can confirm i'm no longer experiencing panics, is there a timeline on release of 3.88.1? I suspect a non-trivial number of folks are hitting this one. From one maintainer to another, thanks for all that you guys do!!

**Comment by kerruba-milkman** [comment]
Facing the same issue on the cli. Rolled back to v3.87.0 solved the issue

**Comment by alshashiguchi** [comment]
I had the same problem using Azure Pipelines when running the task I added input `versionSpec: "3.87.0"`

**Comment by AJHeitzman** [comment]
> I now realize I could have done a better job updating this. This issue seems to appear when there's an empty section in a yaml, that we previously accepted, but we now end up `panic`ing.
> 
> For example something like this:
> 
> ```
> config:
>   aws:region: us-east-1
>   aws:id:
>     -
> ```
> 
> We're working on a fix for this in #14149, but as a workaround in the meantime please look for any such section in your yaml files, and remove those. This should get pulumi back to working.

Confirmed working when I removed the empy sections in the stacks yaml

**Comment by tgummerer** [comment]
pulumi 3.88.1 has been released, and the CLI should no longer choke on this kind of yaml file after upgrading to that.
